modules/root/root_api.rb

```diff
@@ -25,4 +27,19 @@ class Proxy::RootApi < Sinatra::Base
       log_halt 400, e
     end
   end
+
+  get "/events" do
+    begin
+      content_type :json
+
+      events = []
+      from_serial = params[:from_serial].to_i rescue 0
```

Comment:
Nah, every record in the buffer has a serial (increasing integer), which can be used to skip N records when doing additional request (e.g. in one user UI session). From timestamp is not supported, I don't see a use case for that. 

Removed the extra rescue, good point.